# Potencia en un sistema trifásico balanceado

__UNLZ - Facultad de Ingeniería__       
__Electrotecnia__      
__Alumno:__ Daniel Antonio Lorenzo

<a href="https://colab.research.google.com/github/daniel-lorenzo/Electrotecnia/blob/master/Potencia_trifasico.ipynb"><img align="left" src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open in Colab" title="Open and Execute in Google Colaboratory"></a>

## Ejemplo 12.6
Determine los valores totales de la potencia promedio, potencia reactiva y potencia compleja en la fuente y en la carga.

<img src="img/ej12-2.png">

### Solución
Es suficiente considerar una fase, ya que el sistema está balanceado. En relación con la fase $a$.

In [1]:
import math, cmath

In [2]:
# Datos:
Zy = 5-2j + 10+8j
Vp = 110

Ip = Vp/Zy

In [3]:
print('Zy = {:.0f} Ohm'.format(Zy))
print('Ip = (%.2f < %.2f°) A'%(abs(Ip) , math.degrees( cmath.phase(Ip) ) ))

Zy = 15+6j Ohm
Ip = (6.81 < -21.80°) A


$$ V_p = 110 \angle 0^\circ \, \mathrm{V} \quad \mbox{y} \quad I_p = 6,81 \angle -21,8^\circ $$

Así, en la fuente, la potencia compleja suministrada es

$$ S_s = 3 V_p I_p^\ast $$

In [4]:
Ss = 3*Vp*Ip.conjugate()

In [20]:
print('Ss = {:.1f} VA'.format(Ss))

Ss = 2086.2+834.5j VA


En la carga, la potencia compleja absobida es

$$ S_L = 3 |I_p|^2 Z_p $$

In [23]:
Zp = 10+8j

SL = 3*abs(Ip)**2 * Zp

In [24]:
print('SL = {:.1f} VA'.format(SL))

SL = 1390.8+1112.6j VA


La potencia real absorbida es de 1390.8 W y la potencia reactiva absorbida de 1112.6 VAr. La diferencia entre las dos potencias complejas es absorbida por la impedancia de línea $(5-j2) \, \Omega$. Para demostrar que éste es el caso, la potencia compleja absorbida por la línea se halla como

$$ S_\ell = 3|I_p|^2 Z_\ell $$

In [25]:
Zl = 5-2j # Ohm

Sl = 3*abs(Ip)**2 * Zl

In [26]:
print('Sl = {:.1f} VA'.format(Sl))

Sl = 695.4-278.2j VA


la cual es la diferencia entre $S_s$ y $S_L$, como era de esperar.

In [18]:
print('Ss - Sl = {:.1f} VA'.format(Ss - SL) )

Ss - Sl = 695.4-278.2j VA


In [27]:
%reset -s -f

## Problema de práctica 12.6
En referencia al circuito Y-Y del problema de práctica 12.2, calcule la potencia compleja en la fuente y en la carga.

<img src="img/ejp12-2.png">

In [28]:
import math, cmath

In [29]:
# Datos:
Zy = (0.4+0.3j) + (0.6+0.7j) + (24+19j) # Ohm
Zp = 24+19j # Ohm
Vp = cmath.rect(120, math.radians(30) )

In [ ]:
Ip = Vp/Zy

$$ I_p = \frac{V_p}{Z_Y} $$

In [30]:
print('Zy = {:.0f} Ohm'.format(Zy))
print('Ip = (%.2f < %.2f°) A'%(abs(Ip) , math.degrees( cmath.phase(Ip) ) ))

Zy = 25+20j Ohm
Ip = (3.75 < -8.66°) A


In [31]:
# Potencia compleja suministrada
Ss = 3*Vp*Ip.conjugate()

$$ S_s = 3 V_p I_p^\ast $$

In [36]:
print('---------------------')
print('Ss = {:.1f} VA'.format(Ss))
print('---------------------')

---------------------
Ss = 1053.7+842.9j VA
---------------------


In [33]:
# Potencia compleja absorbida
SL = 3*abs(Ip)**2*Zp

$$ S_L = 3 |I_p|^2 Z_p $$

In [37]:
print('---------------------')
print('SL = {:.1f} VA'.format(SL))
print('---------------------')

---------------------
SL = 1011.5+800.8j VA
---------------------


In [38]:
%reset -s -f

## Ejemplo 12.7
Un motor trifásico puede considerarse una carga en Y balanceada. Un motor trifásico toma 5.6kW cuando la tensión de línea es de 220 V y la corriente de línea de 18.2 A. Determine el factor de potencia del motor.

### Solución
La potencia es
$$ S = \sqrt{3} V_L I_L $$

Dado que la potencia real es
$$ P = 5600 \, \mathrm{W} = S \cos \theta $$

el factor de potencia
$$ \mathrm{fp} = \cos \theta = \frac{P}{S} $$

In [40]:
import math, cmath

In [41]:
# Datos
VL = 220 # V
IL = 18.2 # A
P = 5600 # W

In [42]:
S = math.sqrt(3)*VL*IL
fp = P/S

In [45]:
print('Resultados:')
print('S = %.2f VA'%S)
print('fp = %.4f'%fp)

Resultados:
S = 6935.13 VA
fp = 0.8075


In [46]:
%reset -s -f

## Problema de práctica 12.7
Calcule la corriente de línea requerida para un motor de 30kW con un factor de potencia atrasado de 0.85 si se conecta a una fuente balanceada con una tensión de línea de 440 V.

In [47]:
import math, cmath

In [48]:
# Datos:
P = 30e3 # W
fp = 0.85
VL = 440 # V

el factor de potencia
$$ \mathrm{fp} = \cos \theta = \frac{P}{S} $$

Entonces
$$ S = \frac{P}{\mathrm{fp}} $$

In [49]:
S = P/fp

La potencia es
$$ S = \sqrt{3} V_L I_L $$

Entonces
$$ I_L = \frac{S}{\sqrt{3} V_L} $$

In [50]:
IL = S/(math.sqrt(3)*VL)

In [51]:
print('IL = %.2f A'%IL)

IL = 46.31 A


## Ejemplo 12.8
Dos cargas balanceadas se conectan a una línea de 240 kV rms a 60 Hz, como se muestra en la figura. La carga 1 toma 30 kW con un factor de potencia atrasado de 0,6 mientras que la carga 2 toma 45 kVAR con un factor de potencia atrasado de 0,8. Suponiendo la secuencia *abc*, determine:
1. Las potencias compleja, real y reactiva absorbidas por la carga combinada
2. Las corrientes de línea 
3. La capacidad nominal en kVAR de los tres capacitores conectados en $\Delta$ en paralelo con la carga que elevarán el factor de potencia atrasado de 0,9 y la capacitancia de cada capacitor.

### Solución
1. En cuanto a la carga 1, dado que $P_1 = 30 \, \mathrm{kW}$ y $\cos \theta_1 = 0,6$, entonces $\sin \theta_1 = 0,8$. Por lo tanto,

In [9]:
# Datos:
P1 = 30e3 # W
fp1 = 0.6
fq1 = 0.8

In [10]:
import math, cmath

In [11]:
Q1 = (P1/fp1)*fq1
S1 = complex(P1,Q1)

In [14]:
print('Potencia compleja debido a la carga 1:')
print('S1 = {:.1f} kVA'.format(S1/1e3))

Potencia compleja debido a la carga 1:
S1 = 30.0+40.0j kVA


In [19]:
fp2 = 0.8
fq2 = math.sin( math.acos(fp2) )
Q2 = 45e3 # VAR

$\mathrm{fp}_2 = \cos \theta_2 = 0,8 \quad \rightarrow \quad \mathrm{fq}_2 = \sin \theta_2 = 0,6$

$$ |S_2| = \frac{Q_2}{\sin \theta_2} $$

$$ P_2 = |S_2| \cos \theta_2 $$

In [20]:
P2 = (Q2/fq2)*fp2
S2 = complex(P2,Q2)

In [23]:
print('S2 = {:.1f} kVA'.format(S2/1e3))

S2 = 60.0+45.0j kVA


In [24]:
# Potencia total absorbida
S = S1 + S2

In [27]:
print('S = {:.1f} kVA'.format(S/1e3))
print('S = (%.1f < %.2f°) kVA'%(abs(S/1e3) , math.degrees( cmath.phase(S) ) ))

S = 90.0+85.0j kVA
S = (123.8 < 43.36°) kVA


In [28]:
fp = math.cos( cmath.phase(S) )

In [29]:
print('fp = %.3f'%fp)

fp = 0.727


In [37]:
# 2. item
V_lin = 240e3 # V
I_lin = (S/(math.sqrt(3)*V_lin)).conjugate()
Ia = I_lin
Ib = Ia * cmath.rect(1 , math.radians(-120) )
Ic = Ia * cmath.rect(1 , math.radians( 120) )

In [38]:
print('I_lin = (%.2f < %.2f°) mA'%(abs(I_lin*1e3), math.degrees( cmath.phase(I_lin) ) ))
print('---------------------------------')
print('Ia = (%.2f < %.2f°) mA'%(abs(Ia*1e3), math.degrees( cmath.phase(Ia) ) ))
print('Ib = (%.2f < %.2f°) mA'%(abs(Ib*1e3), math.degrees( cmath.phase(Ib) ) ))
print('Ic = (%.2f < %.2f°) mA'%(abs(Ic*1e3), math.degrees( cmath.phase(Ic) ) ))

I_lin = (297.80 < -43.36°) mA
---------------------------------
Ia = (297.80 < -43.36°) mA
Ib = (297.80 < -163.36°) mA
Ic = (297.80 < 76.64°) mA


In [43]:
# 3. ítem
theta_i = cmath.phase(S)
theta_f = math.acos(0.9)
P = S.real

Qc = P*(math.tan(theta_i) - math.tan(theta_f)) 

In [44]:
print('Qc = %.1f kVAR'%(Qc/1e3))

Qc = 41.4 kVAR


In [45]:
Qc2 = 13800 # VAR
f = 60 # Hz
w = 2*math.pi*f

C = Qc2/(w*V_lin**2)

In [48]:
print('C = %.1f pF'%(C*1e12))

C = 635.5 pF


In [49]:
%reset -s -f

## Ejemplo 12.9
La carga en Y desbalanceada de la figura tiene tensiones balanceadas de 100 V y la secuencia *abc*. Calcule las corrientes de línea y la corriente neutra. Considere $Z_A = 15 \, \Omega, \; Z_B = 10 + j5 \, \Omega, \; Z_C = 6 - j8 \, \Omega$.

### Solución
Las corrientes de línea son

In [50]:
import math, cmath

In [51]:
# Datos:
Va = cmath.rect(100 , 0)
Vb = cmath.rect(100 , math.radians(120))
Vc = cmath.rect(100 , math.radians(240))
Za = 15    # Ohm
Zb = 10+5j # Ohm
Zc =  6-8j # Ohm

In [52]:
# Corrientes de línea
Ia = Va/Za
Ib = Vb/Zb
Ic = Vc/Zc

In [53]:
print('Ia = (%.2f < %.2f°) A'%(abs(Ia) , math.degrees( cmath.phase(Ia) ) ) )
print('Ib = (%.2f < %.2f°) A'%(abs(Ib) , math.degrees( cmath.phase(Ib) ) ) )
print('Ic = (%.2f < %.2f°) A'%(abs(Ic) , math.degrees( cmath.phase(Ic) ) ) )

Ia = (6.67 < 0.00°) A
Ib = (8.94 < 93.43°) A
Ic = (10.00 < -66.87°) A


In [56]:
# Corriente de línea neutra
In = -(Ia + Ib + Ic)

In [57]:
print('In = (%.2f < %.2f°) A'%(abs(In) , math.degrees( cmath.phase(In) ) ) )

In = (10.06 < 178.47°) A


In [58]:
%reset -s -f

-----------
<a href="https://colab.research.google.com/github/daniel-lorenzo/Electrotecnia/blob/master/Potencia_trifasico.ipynb"><img align="left" src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open in Colab" title="Open and Execute in Google Colaboratory"></a>